eth/stagedsync/stage_execute.go

```diff
@@ -9,6 +9,7 @@ import (
 	"time"
 
 	"github.com/c2h5oh/datasize"
+	lru "github.com/hashicorp/golang-lru"
```

Comment:
you have global lock in mutation - then you don't need one more lock inside lru (probably), then you can use `simplelru` sub-package.

Reply:
there is also github.com/VictoriaMetrics/fastcache package which can be more GC-friendly, but I'm not sure.